File: data/fighter.lua
local fighter = {}

fighter.baseHATK = 5
fighter.baseMATK = 3
fighter.baseLATK = 1

function fighter.create()
	local newFighter = {
		health = 100,
		focus = 50,
		speed = math.random(1, 5),
		strength = math.random(1, 5),
		defense = math.random(1, 5),
		specialCost = 100, -- TODO: multiple slotting rework
	}
	return newFighter
end

return fighter

File: data/fighter.c
#include "fighter.h"
#include <stdlib.h>
#include <time.h>

int speed_max = 5;
int strength_max = 5;
int defense_max = 5;

Fighter initFighter() {
  Fighter newFighter = {
    health: 100,
    stamina: 50
  };

  newFighter.speed = rand() % speed_max;
  newFighter.strength = rand() % strength_max;
  newFighter.defense = rand() % defense_max;
  return newFighter;
};
